### PR TITLE
Update Site To 2021 Data

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
Created and added new data from 2021, with a new data process that combines historic data with the 2021 data, allowing buildings with only past data like the Art Institute to still show useful information..

**⚠️  Warning** - this is currently a destructive update, since you cannot see 2020 data on buildings that reported their 2021 data. We should likely follow this up with a PR tackling historical emissions (Issue #4).

### Details

The new data:
```src/data/source/ChicagoEnergyBenchmarking.csv``` is the source file
```src/data/source/ChicagoEnergyBenchmarkingAllNewestInstances.csv``` are all the newest unique building ID's
```src/data/source/ChicagoEnergyBenchmarkingThisYear.csv``` is all submitted data for the current year (2021)

When displaying, an alert will pop up if the webpage is using old (not 2021) data:
image

## Known Issues To Resolve

- [x] Cleanest Buildings Page Column Data is Missing
- [x] Some Building Titles Now Start with `(IL###) - `
- [x] Images are Disconnected From Their Respective Buildings Due to Slug Changes 
- [x] Stats on Clean Buildings (e.g. Marina Towers) Are Weird
- [x] Building Owners are Disconnected From Their Respective Buildings Due to Slug Changes (switched to ID!)
- [x] All Buildings Now Report Gas, District Steam, & District Chilling - 0 Values Get Ranked & Rendered
- [x] District Steam & Cooling Don't Show Median (added note, we'd need to calculate this filtered by non-zero buildings)
- [x] Add Release Notes to Indicate We Updated Data to 2021

### CODE REVIEW

**Comment by vkoves** [comment]
Haven't reviewed this yet, but it gives me a chuckle that IIT is still #1 😆 

image

We need to tag Crown Hall as an IIT property, and maybe shift to using building ID instead of slugs for owners, since that's probably more stable year over year

**Comment by vkoves** [comment]
@SuragNuthulapaty - ope, I found a bug. We show buildings that _never submitted_, like the [United Center](https://deploy-preview-40--radiant-cucurucho-d09bae.netlify.app/building/united-center/) but the page doesn't work because there's no data:

https://deploy-preview-40--radiant-cucurucho-d09bae.netlify.app/building/united-center/

I think changing the yellow warning to a red warning and saying they never submitted then hiding all the info we don't have should be sufficient, and that will solve Issue #25

**Comment by vkoves** [comment]
Oh one more thing - I think old buildings are getting included in rankings, like "Hyde Park West Apartments" on the homepage. This shouldn't happen, since it isn't really fair to compare buildings in separate years where the data collection could have been different or even just building standards have changed.

**Comment by SuragNuthulapaty** [comment]
@vkoves This is the raw csv file:
Screenshot 2023-07-14 at 10 20 49 PM
I see that united center has submitted data, I am not sure why it's not working. In that row all the data is there as normal.

**Comment by vkoves** [comment]
Alright going to try digging into some of these data quirks. Turns out for some reason now buildings sometimes get prefixed with `(ILNUMBER) - `, so we should try and trim that out:

Screenshot from 2023-07-18 20-00-45

**Comment by vkoves** [comment]
So thinking about this more, I think the negative ranking is happening because we are running the add context by property type ranking _before_ the data split by year, which likely means we are ranking across _all buildings across all years_, so Marina Towers may be ranked as #1,999 dirtiest of #2,000 buildings in the raw data set, but then there's only say #1,000 buildings in the latest year, so the rank gets inverted to 1000 - 1999 = -999. We should likely only rank buildings against buildings in the latest year to simplify this.

File: README.md [hunk 1]
```diff
@@ -90,4 +88,6 @@ cd src/data
 pip install --no-cache-dir -r requirements.txt
 python3 scripts/process-data.py
 python3 scripts/add-context-by-property-type.py
+python3 scripts/clean-and-pare-down-data_all_years.py
```
Comment (vkoves) [inline README.md:91]: @SuragNuthulapaty - are you sure this order is correct? Also the new files are in the source directory, not the dist directory, which means they aren't actually being used for the main site itself, just `building-benchmarks.csv`. Is that correct?

**Comment by SuragNuthulapaty** [comment]
Just took a look and i believe I got it wrong. The ```python3 scripts/clean-and-pare-down-data_all_years.py``` should be on line 89, not 91, with two lines above it being brought down. Also, the reverence to ```building-benchmarks.csv``` is correct, the next two scropts will correctlyupdate the data in ```./dist/``` to use on the main site.

**Comment by derekeder** [comment]
Marina Towers is now reporting zero values for gas use, so the front-end rankings calculation needs to account for this:

Screen Shot 2023-08-22 at 9 59 26 PM

source data (2021 at the top)
Screen Shot 2023-08-22 at 10 03 58 PM

**Comment by vkoves** [comment]
District cooling and heating no longer shows medians, a good example is Herman Hall.

| Prod | This Branch |
| --- | --- |
| Screenshot from 2023-08-22 23-29-40 | Screenshot from 2023-08-22 23-29-33 |

**Comment by vkoves** [comment]
Large owners and building images are now fixed with a pivot to IDs! 🎉 

Screenshot from 2023-08-24 22-16-06

**Comment by vkoves** [comment]
I've updated the natural gas tile to provide more information on a building's gas use based on whether or not it uses district steam! We may want to move this up later on, since we should really highlight the few fully gas free buildings:

| Fully Gas Free (e.g. Marina Towers) | No Local Gas (e.g. Crown Hall) |
| --- | --- |
| Screenshot from 2023-08-27 22-49-31 | Screenshot from 2023-08-27 22-49-26 |

File: src/data/scripts/add-context-by-property-type.py [hunk 2]
```diff
@@ -27,30 +27,39 @@
...
+# TODO: output property_types to a json file for use in the frontend
+
 def calculateBuildingStatistics():
     stats_by_property_type = {}
-    all_property_types = pd.read_json("./dist/property-types.json")
 
     # looping through both all the property types and all the columns we want to get data on
-    for i, property in enumerate(all_property_types["propertyTypes"]):
+    for i, property in enumerate(property_types):
+        print("property", property)
```
Comment (derekeder) [inline src/data/scripts/add-context-by-property-type.py:56]: remove this print statement

File: src/data/scripts/building_utils.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+###
+### Building Utils - Common utilities for processing building data
```
Comment (derekeder) [inline src/data/scripts/building_utils.py:2]: could this file be combined with `utils.py`?

**Review by derekeder — APPROVED** [review]
A few cleanup notes, but otherwise this is good to go

**Pull request merged**